File: java/03-bootstrap/src/main/java/com/fResult/bootstrap/customers/services/DevelopmentOnlyCustomerService.java
package com.fResult.bootstrap.customers.services;

import com.fResult.bootstrap.common.utils.DataSourceUtils;
import javax.sql.DataSource;
import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;

public class DevelopmentOnlyCustomerService extends BaseCustomerService {
  public DevelopmentOnlyCustomerService() {
    super(buildDataSource());
  }

  private static DataSource buildDataSource() {
    final var dataSource = new EmbeddedDatabaseBuilder().setType(EmbeddedDatabaseType.H2).build();

    return DataSourceUtils.initializeDdl(dataSource);
  }
}

File: kotlin/03-bootstrap/src/main/kotlin/com/fResult/bootstrap/customers/services/DevelopmentOnlyCustomerService.kt
package com.fResult.bootstrap.customers.services

import com.fResult.bootstrap.common.utils.DataSourceUtils
import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder
import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType
import javax.sql.DataSource

class DevelopmentOnlyCustomerService : BaseCustomerService(buildDataSource()) {
  companion object {
    private fun buildDataSource(): DataSource {
      val dataSource = EmbeddedDatabaseBuilder().setType(EmbeddedDatabaseType.H2).build()

      return DataSourceUtils.initializeDdl(dataSource)
    }
  }
}
